src/main/scala/tech/cryptonomic/conseil/Lorre.scala

```diff
@@ -9,12 +9,23 @@ import scala.concurrent.Await
 import scala.concurrent.ExecutionContext.Implicits.global
 import scala.concurrent.duration.Duration
 import scala.util.{Failure, Success, Try}
+import scala.annotation.tailrec
 
 /**
   * Entry point for synchronizing data between the Tezos blockchain and the Conseil database.
   */
 object Lorre extends App with LazyLogging {
 
+  private val network =
```

Comment:
Look up the network validation pull request. There's logic for this here you can reuse/improve if needed. We might scrap that pr and use whatever is useful/best practice here and elsewhere

Reply:
I did add my own comments on PR #109 for consideration.

I will not change the code here, and depending on which PR got merged first we will overwrite the code in this branch or the other with the one from #109